src/script/modules/get-cof-value/index.ts

```diff
@@ -0,0 +1,66 @@
+import settings from '../../lib/settings';
+import { getCofStore } from '../../utils/snapchat';
+
+let oldGetClientCofValue: any = null;
+let oldGetClientCofValueAndLogExposure: any = null;
+
+function handleStoreEffect(storeState: any) {
+
+  if (oldGetClientCofValue == null) {
+    oldGetClientCofValue = storeState.getClientCofValue;
+    storeState.getClientCofValue = function (...args: any[]) {
+      const mobileEnabled = settings.getSetting('SNAP_AS_MOBILE');
+      if (mobileEnabled && args[0] && args[0] === 'DWEB_SNAP_SENDING_CONTEXT') {
+        return true;
+      }
+      const adsEnabled = settings.getSetting('ADS_ENABLED');
+      if (args[0] && args[0] === 'DWEB_ENABLE_ADS') {
+        return adsEnabled ? "enabled" : "disabled";
+      }
+      const myaiEnabled = settings.getSetting('MY_AI_MENTIONS');
+      if (args[0] && args[0] === 'DWEB_ENABLE_MY_AI_MENTION') {
+        return myaiEnabled ? "enabled" : "disabled";
+      }
+      const privStoryEnabled = settings.getSetting('PRIVATE_STORIES');
+      if (args[0] && args[0] === 'DWEB_PRIVATE_STORIES_VIEWING') {
```

Comment:
You check `args[0]` is defined several times here, maybe do it once before all these checks?